src/shared/components/FileStep/FileStep.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react'
+import Icon from '../Icon'
+import {
+  StepWrapper,
+  StepStatus,
+  StepNumber,
+  StepDetails,
+  DeleteButton,
+  FileType,
+  FileName,
+  Thumbnail,
+  StyledProgress,
+} from './FileStep.style'
+import { Step } from '../MultiFileSelect'
+
+export type FileStepProps = {
+  stepNumber: number
+  active: boolean
+  onDelete: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
+  step: Step
+  fileName?: string
+  thumbnail?: string
+  onChangeStep?: (step: Step) => void
```

Comment:
I think this more of a `onSelect` right? The consumer ultimately decides if the step is changed